docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+## Implementing Links
+
+Before implementing your own Link, be sure to read the [usage](usage.md) document.
+
+Links implement a `request` method that takes an operation and returns the results in an Observable.
+The apollo-link's `Observable` follows the ECMAScript [proposal](https://github.com/tc39/proposal-observable#api) with some extensions provided by [zen-observable](https://github.com/zenparsing/zen-observable#api).
+
+### Creating a Simple Http Link
+
+All links extend the base class, `ApolloLink`, which contains the methods for composition as well as the signature of request.
```

Comment:
You could take the builder approach, if chaining is preferable.  Something like:

```ts
const link = ApolloLink
  .chain(loggingLink)
  .split(
    isLocal,
    checkLocalCache,
    ApolloLink.passthrough(),
  )
  .concat(dedup)
  .concat(retry)
  .concat(http)
  .link();
```